### PR TITLE
Cc license submission step

### PR DESCRIPTION
**Pull request by samuelcambien**
## References

https://github.com/DSpace/Rest7Contract/blob/master/submissioncclicenses.md

## Description

This PR implements the CC licenses step of the submission.

## Instructions for Reviewers

* Added models, resource types and data services for 'SubmissionCcLicence' and 'SubmissionCcLicenceUrl'
* Added a new component 'SubmissionSectionCcLicensesComponent' which extends SectionModelComponent to render the CcLicense submission section
* Added a new component 'DsSelectComponent' for rendering dropdowns. This was extracted from the SubmissionFormCollectionComponent.
* Added a new interface 'WorkspaceitemSectionCcLicenseObject' to store the CC licenses section data

Expected behaviour:

* The submission CC licenses section should be added to the submission form
* It should show a dropdown with possible license types
* When selecting a type and answering the questions (if applicable), the link to the licenss should appear, along with a confirmation checkbox
* When confirming and saving the submission, this confirmation should be stored in the submission along with the link

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs for any bug fixes, improvements or new features. A few reminders about what constitutes good tests:
    * Include tests for different user types (if behavior differs), including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
    * Include tests for error scenarios, e.g. when errors/warnings should appear (or buttons should be disabled).
    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/submission/form/collection/submission-form-collection.component.html [hunk 1]
```diff
@@ -1,50 +1,44 @@
-<div>
```
Comment (atarix83) [inline src/app/submission/form/collection/submission-form-collection.component.html:1]: Please consider there is a [new PR](https://github.com/DSpace/dspace-angular/pull/708) that introduce a refactoring of this component. I should revert these changes because will generate a conflicts unnecessary

File: src/app/shared/submission-section-cc-licenses/submission-section-cc-licenses.component.html [hunk 2]
```diff
@@ -0,0 +1,143 @@
+<div class="mb-4 ccLicense-select">
+  <ds-select
+    [disabled]="!submissionCcLicenses"
+    [label]="'submission.sections.ccLicense.type'">
+
+    <ng-container class="selection">
+      <span *ngIf="!submissionCcLicenses">
+        <i class='fas fa-circle-notch fa-spin'></i>
```
Comment (atarix83) [inline src/app/shared/submission-section-cc-licenses/submission-section-cc-licenses.component.html:8]: Why don't you use the `ds-loading` component in order to conform to the way dspace displays the loading state

File: src/app/shared/submission-section-cc-licenses/submission-section-cc-licenses.component.html [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    </ng-container>
+
+  </div>
+
+</ng-container>
+
+<ng-container *ngIf="getCcLicenseLink$()">
+  <ng-container *ngVar="getCcLicenseLink$() | async as licenseLink">
+    <div *ngIf="!licenseLink">
+      <i class='fas fa-circle-notch fa-spin m-2'></i>
```
Comment (atarix83) [inline src/app/shared/submission-section-cc-licenses/submission-section-cc-licenses.component.html:123]: same as above

**Comment by samuelcambien** [comment]
Hi Tim and Giuseppe, thanks for your feedback.
The error with the cc license request was caused by a change in the REST contract, implemented by this PR:

https://github.com/DSpace/DSpace/pull/2764

The code isn't consistent with the REST contract though:
* endpoint: 'config/submissioncclicenseurls/search/rightsByQuestions'
* self link: 'config/submissioncclicenses/search/rightsByQuestions'
* actual working endpoint: 'config/submissioncclicenseUrl/search/rightsByQuestions'

In the angular code I used the id 'submissioncclicenseUrl-search' and now it's working.
I had to overwrite the getSearchEndpoint method to get rid of the hardcoded '/search' addition, since this is already included in the endpoint coming from the HalService.

Regarding the double 'CC0' licence type: I don't know what it means, but it is returned like this by the 'config/submissioncclicenses' endpoint.

**Comment by tdonohue** [comment]
@samuelcambien : It looks like this is now failing tests in Travis CI.  At a glance, it looks like you may have only updated the code and forgot to update the corresponding tests.

Also, this has a minor merge conflict now (I just merged #708 which seems to have created this merge conflict).  So, it'll need a quick rebase as well.

**Comment by samuelcambien** [comment]
Hi Tim, sorry I missed the test failures indeed, fixed them now and merged the latest master.
